src/common/dcsvg.cpp

```diff
@@ -108,6 +118,151 @@ wxString wxBrushString(wxColour c, int style = wxBRUSHSTYLE_SOLID)
     return s;
 }
 
+wxString wxGetPenPattern(wxPen& pen)
+{
+    wxString s;
+    float w = pen.GetWidth();
+    if (pen.GetWidth() == 0)
+        w = 1;
+    w = w / 3;
+
+    switch (pen.GetStyle())
+    {
+        case wxPENSTYLE_DOT:
+            s.Printf(wxT("stroke-dasharray=\"%f,%f\" "), w * 2, w * 5);
+            break;
+        case wxPENSTYLE_SHORT_DASH:
+            s.Printf(wxT("stroke-dasharray=\"%f,%f\" "), w * 10, w * 8);
+            break;
+        case wxPENSTYLE_LONG_DASH:
+            s.Printf(wxT("stroke-dasharray=\"%f,%f\" "), w * 15, w * 8);
+            break;
+        case wxPENSTYLE_DOT_DASH:
+            s.Printf(wxT("stroke-dasharray=\"%f,%f,%f,%f\" "), w * 8, w * 8, w * 2, w * 8);
+            break;
+        case wxPENSTYLE_USER_DASH:
+        {
+            s = wxT("stroke-dasharray=\"");
+            wxDash *dashes;
+            int count = pen.GetDashes(&dashes);
+            if ((dashes != NULL) && (count > 0))
+            {
+                for (int i = 0; i < count; ++i) {
+                    s << dashes[i];
+                    if (i < count - 1)
+                        s << ",";
+                }
+            }
+            s += wxT("\" ");
+            break;
+        }
+        default:
```

Comment:
Just to make this more clear: it's a good idea to write exhaustive switch statements over enums, i.e. take into account all possible enum values. So in this case I'd rather all of `case wxPENSTYLE_XXX_HATCH:` explicitly here, even if we don't handle them and also add `case wxPENSTYLE_INVALID:` even if it can't really occur here (or can it? I'm not even sure, the function doesn't document its preconditions) and remove `default:`.

The advantage is that if some new `wxPENSTYLE_ZZZ` is added later, the compiler will remind us to add support for it here too.